tests/test.e2e.js

```diff
@@ -1,16 +1,63 @@
 import waitPort from 'wait-port'
 import { remote } from 'webdriverio'
 
-import { start } from '../dist/index.js'
+import { download, start } from '../dist/index.js'
+
+// start geckodriver automatically
+console.log('= start geckodriver automatically =')
+try {
+  const browser = await remote({
+    automationProtocol: 'webdriver',
+    capabilities: {
+      browserName: 'firefox',
+      'moz:firefoxOptions': {
+        args: ['-headless']
+      }
+    }
+  })
+  await browser.url('https://webdriver.io')
+  await browser.deleteSession()
+} catch (err) {
+  console.error(err)
+  process.exit(1)
+}
+
+// start specific geckodriver
+console.log('= start specific geckodriver =')
+
+const binary = await download()
+
+try {
+  const browser = await remote({
+    automationProtocol: 'webdriver',
+    capabilities: {
+      browserName: 'firefox',
+      'moz:firefoxOptions': {
+        args: ['-headless']
+      },
+      'wdio:geckodriverOptions': {
+        binary
+      }
+    }
+  })
+  await browser.url('https://webdriver.io')
+  await browser.deleteSession()
+} catch (err) {
+  console.error(err)
+  process.exit(1)
+}
+
+// start geckodriver manually
+console.log('= start geckodriver manually =')
 
 const port = 4444
 const cp = await start({ port })
 
 try {
-  await waitPort({ port: 4444 })
+  await waitPort({ port })
   const browser = await remote({
     automationProtocol: 'webdriver',
-    hostname: '0.0.0.0',
```

Comment:
Note that this is no longer necessary as the default value has changed from `localhost` to `0.0.0.0`. See: https://github.com/webdriverio/webdriverio/commit/c6d81b5e40f43b720b0dffd1adbd9985badf35ca#diff-b5fe71260eef292213595160b7583dabd5475f9775eee92b92d00d0c5fb1fa7f